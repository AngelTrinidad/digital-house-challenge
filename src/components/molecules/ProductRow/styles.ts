import {StyleSheet} from 'react-native';

import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: 'white',
    gap: scaleSpace(12),
    minHeight: scaleSpace(56),
    alignItems: 'center',
  },
  image: {
    borderRadius: 10,
    width: scaleSpace(55),
    height: scaleSpace(55),
  },
  pointContainer: {
    flexDirection: 'row',
    alignItems: 'flex-end',
  },
  date: {
    marginTop: scaleSpace(8),
    fontWeight: '400',
  },
});

export default styles;
