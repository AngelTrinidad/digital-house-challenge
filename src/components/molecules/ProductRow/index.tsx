import React, {FC, memo} from 'react';
import {Image, TextStyle, View} from 'react-native';

import styles from './styles';

import {Typography, ViewFlex} from '~components/atoms';
import {useTheme} from '~config/themes';
import {formatNumber, getFullDate} from '~helpers';
import {Product} from '~interfaces';

interface Props {
  data: Product;
}

const ProductRow: FC<Props> = ({
  data: {createdAt, image, product, is_redemption, points},
}) => {
  const {colors} = useTheme();
  const humanizedDate = getFullDate(createdAt);
  const prefixStyles: TextStyle = {
    color: is_redemption ? colors.error : colors.success,
  };

  return (
    <View style={styles.container}>
      <Image source={{uri: image}} style={styles.image} resizeMode="cover" />
      <ViewFlex>
        <Typography numberOfLines={2}>{product}</Typography>
        <Typography variant="sm" style={styles.date}>
          {humanizedDate}
        </Typography>
      </ViewFlex>
      <View style={styles.pointContainer}>
        <Typography variant="lg" style={prefixStyles}>
          {is_redemption ? '-' : '+'}
        </Typography>
        <Typography variant="lg" adjustsFontSizeToFit>
          {formatNumber(points)}
        </Typography>
      </View>
      <Typography variant="lg">{'>'}</Typography>
    </View>
  );
};

export default memo(ProductRow);
