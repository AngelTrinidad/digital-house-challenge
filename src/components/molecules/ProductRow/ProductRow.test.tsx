import React from 'react';

import ProductRow from '.';

import {render} from '~helpers/testing';
import {product} from '~mocks';

describe('<ProductRow />', () => {
  let component: any;

  beforeEach(() => {
    component = render(<ProductRow data={product} />);
  });

  it('should renders correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has the same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
