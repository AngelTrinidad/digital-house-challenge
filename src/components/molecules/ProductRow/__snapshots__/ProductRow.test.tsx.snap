// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductRow /> should has the same snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flexDirection": "row",
      "gap": 13.714285714285715,
      "minHeight": 64,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      {
        "uri": "https://loremflickr.com/640/480/transport",
      }
    }
    style={
      {
        "borderRadius": 10,
        "height": 62.857142857142854,
        "width": 62.857142857142854,
      }
    }
  />
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
        undefined,
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={
        {
          "color": "black",
          "fontFamily": "Metropolis Bold",
          "fontSize": 17.2,
          "fontWeight": "800",
        }
      }
    >
      Handmade Metal Shoes
    </Text>
    <Text
      style={
        {
          "color": "black",
          "fontFamily": "Metropolis",
          "fontSize": 14.742857142857144,
          "fontWeight": "400",
          "marginTop": 9.142857142857142,
        }
      }
    >
      9 de Diciembre, 2022
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        {
          "color": "#00B833",
          "fontFamily": "Metropolis Bold",
          "fontSize": 19.65714285714286,
          "fontWeight": "800",
        }
      }
    >
      +
    </Text>
    <Text
      adjustsFontSizeToFit={true}
      style={
        {
          "color": "black",
          "fontFamily": "Metropolis Bold",
          "fontSize": 19.65714285714286,
          "fontWeight": "800",
        }
      }
    >
      20,000
    </Text>
  </View>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": "Metropolis Bold",
        "fontSize": 19.65714285714286,
        "fontWeight": "800",
      }
    }
  >
    &gt;
  </Text>
</View>
`;
